### PR TITLE
Expose reviewers validation result API through JWT auth and document it

### PR DESCRIPTION
**Pull request by diox**
Fixes mozilla/addons#14833

## Context

That endpoint existed already but wasn't documented and was only available through the SessionID auth, which is the default, and not through JWT auth. That's because it was only needed for code-manager until now, which uses session auth. It's now needed for [Assay](https://github.com/mozilla/assay) which uses JWT so I've added the relevant authentication classes.

## Testing

See unit tests, we have used a similar method of duplicating the class and changing the `client_class` for other API tests, makes things easier even it's a little wasteful.

It's painful to test locally and I would just trust the unit tests, but for anyone adventurous, you'd need to:
- Upload an add-on, then grab the add-on and file id (from a shell is easier, i.e. `Addon.objects.latest('pk').pk` and `File.objects.latest('pk').pk`)
- Log in as a reviewer, generate API keys in devhub (need to look up fake emails to grab the confirmation link)
- Using those API keys to generate a JWT-authenticated request against `http://olympia.test/api/v5/reviewers/addon/{addon_id}/file/{file_id}/validation/` following our API docs

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Build Docker image
+        if: "!github.event.pull_request.head.repo.fork"
```
Comment (KevinMind) [inline .github/workflows/ci.yml:24]: rebase.

File: docs/topics/api/reviewers.rst [hunk 2]
```diff
@@ -10,7 +10,8 @@ Reviewers
     See :ref:`the API versions available<api-versions-list>` for alternatives
     if you need stability.
     The only authentication method available at
-    the moment is :ref:`the internal one<api-auth-internal>`.
+    the moment is :ref:`the internal one<api-auth-internal>`, except for
```
Comment (KevinMind) [inline docs/topics/api/reviewers.rst:13]: This is a confusing sentence. There are two auth methods, but the sentence starts with "the only one available" can you rephrase this?

File: src/olympia/reviewers/views.py [hunk 3]
```diff
@@ -1197,6 +1201,10 @@ def clear_pending_rejections(self, request, **kwargs):
         detail=True,
         methods=['get'],
         permission_classes=[AllowAnyKindOfReviewer],
+        authentication_classes=[
```
Comment (KevinMind) [inline src/olympia/reviewers/views.py:1204]: Have you explored what this would look like in the swagger docs? I know we kind of dropped the project of migrating api docs to swagger but could be an interesting exercise to add some anotations to the view class to see if you can explain what you do in the .rst file directly in the swagger.

**Comment by KevinMind** [comment]
Curious @diox did you go through the steps you outlined in the test section? If so, I would be curious to try (another adventurous thing). What if you exported the database, which should include the addon, the files, the session etc and zipped it.

I could try to load that state onto my local and verify without having to repeat the steps. Could be fun. Don't have to, but would be curious what happens.

File: docs/topics/api/reviewers.rst (hunk 2 above)
Comment (diox) [reply docs/topics/api/reviewers.rst:13]: Maybe 

```
The only authentication method available for these APIs is
:ref:`the internal one<api-auth-internal>`, except for the
:ref:`validation results<reviewers-validation>` endpoint, which allows both
internal and :ref:`external auth<api-auth>`.
```

? Ultimately internal auth remains the only auth mechanism available for the others APIs described in that doc, validation is the exception...

File: src/olympia/reviewers/views.py (hunk 3 above)
Comment (diox) [reply src/olympia/reviewers/views.py:1204]: I haven't. It's a bad one for swagger from the start because it doesn't use a serializer and directly returns a `JsonResponse`, so it doesn't automatically document the response (which admittedly only contains a single field). Furthermore it looks like our swagger implementation is not smart enough to know which authentication classes are available for each endpoint, so it was already suggesting to use JWT for that endpoint...

I could add a docstring though.

**Comment by diox** [comment]
> Curious @diox did you go through the steps you outlined in the test section?

I didn't, I was trusting the unit tests (and the fact that this is the standard way of adding JWT auth to our APIs). I shared the PR with @chrstinalin on slack so she could test locally with her build of assay that required this change.

**Review by KevinMind — APPROVED** [review]
Didn't test locally.
